beamer/agent.py

```diff
@@ -100,6 +107,7 @@ def stop(self) -> None:
         self._event_processor.stop()
         self._event_monitor_l2a.stop()
         self._event_monitor_l2b.stop()
+        self.resolution_pool.shutdown(wait=True, cancel_futures=False)
```

Comment:
Why do set `cancel_futures` to `False` here?

Reply:
What troubles me here is the fact that that can potentially take a long time. The idea behind `stop` was to stop regardless of the state we're in.